### PR TITLE
Feature/consolidation

### PR DESCRIPTION
**Pull request by plummip**
- consolidation of PDF papers and raw binary data. less then 50MB base64 output.
https://github.com/entdec/papyrus/issues/9

### CODE REVIEW

**Review by shubhrathasetty — COMMENTED** [review]
Looks good to me

File: Gemfile [hunk 1]
```diff
@@ -18,6 +18,8 @@ gemspec
 # gem 'auxilium', '~> 3', entdec: 'auxilium'
 gem 'decoro', '~> 0.1', entdec: 'decoro'
 gem 'evento', entdec: 'evento', tag: '0.1.7'
+gem 'combine_pdf', '~> 1.0.23'
+gem 'praesens', '~> 0.1', entdec: 'praesens'
```
Comment (tdegrunt) [inline Gemfile:22]: As discussed, praesens might not be needed, it can be, but do we want to?

**Comment by plummip** [comment]
Please update the revision and add it to OMS

**Review by tdegrunt — COMMENTED** [review]
Are batches needed?

File: app/jobs/papyrus/sidekiq_consolidation_job.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module Papyrus
+  class SidekiqConsolidationJob < ApplicationJob
+
+    def perform(consolidation_id)
+      Papyrus.start_consolidation(consolidation_id)
+
+      batch = Sidekiq::Batch.new
```
[the remark this answers is not in the document]
Comment (plummip) [reply app/jobs/papyrus/sidekiq_consolidation_job.rb:7]: No we could also add a redis client that increments and decrements the count on a certain key based on the consolidation ID. if the key count reaches 0 it should be printed by the job.

I think.

**Pull request merged**